java/idl2jni/runtime/idl2jni_runtime.h

```diff
@@ -89,7 +89,7 @@ void copyToCxx(JNIEnv *jni,
 template <typename Traits>
 void copyToJava(JNIEnv *jni, jobject &target,
   const TAO::details::object_reference_const_sequence_element<Traits> &source,
-  bool createNewObject = false)
+  bool /* createNewObject = false */)
```

Comment:
I think this was supposed to match another usage or something that the idlj2ni compiler generates.  We can avoid the warning with `bool /*createNewObject*/ = false)`

But if everything works with your change it may be that it's just never called with 3 parameters.